### PR TITLE
[Merged by Bors] - TY-2979 move feed logic to rust (0): nextFeedBatch

### PR DESCRIPTION
**Pull request by janpetschexain**
**References**

- [TY-2979]
- followed by #518

**Summary**

- add `Engine::feed_next_batch()` & corresponding ffi
- use new logic & remove corresponding db calls in `FeedManager.nextFeedBatch()` in dart

[TY-2979]: https://xainag.atlassian.net/browse/TY-2979?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine/lib/src/domain/feed_manager.dart [hunk 1]
```diff
@@ -109,6 +111,31 @@ class FeedManager {
 
   /// Obtain the next batch of feed documents and persist to repositories.
   Future<EngineEvent> nextFeedBatch() async {
+    if (cfgFeatureStorage) {
+      final sources = await _sourceReactedRepo.fetchAll();
+      final List<DocumentWithActiveData> docs;
```
Comment (biernacki) [inline discovery_engine/lib/src/domain/feed_manager.dart:116]: might be better to mark it as `late final List<DocumentWithActiveData> docs;`, but if the analyzer doesn't complain, then maybe that's fine too 🤔

**Review by biernacki — APPROVED** [review]
lgtm, just one small suggestion, but optional if the analyzer is happy with the current implementation

File: discovery_engine/lib/src/domain/feed_manager.dart (hunk 1 above)
Comment (janpetschexain) [reply discovery_engine/lib/src/domain/feed_manager.dart:116]: if i understood this correctly, then `late final` does the check for initialization at runtime, whereas just declaring the variable (as `final`, and to have it available in the outer scope) and then assigning it in every if-branch (or returning early) does the check at compile time.

**Comment by janpetschexain** [comment]
bors merge

**Pull request closed without merging**